zebra-consensus/src/verify/block.rs

```diff
@@ -471,4 +472,37 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    #[spandoc::spandoc]
+    async fn header_solution() -> Result<(), Report> {
+        install_tracing();
+
+        // Service variables
+        let state_service = Box::new(zebra_state::in_memory::init());
+        let mut block_verifier = super::init(state_service);
+        let ready_verifier_service = block_verifier.ready_and().await.map_err(|e| eyre!(e))?;
+
+        // Get a valid block
+        let mut block =
+            Block::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_415000_BYTES[..])
+                .expect("block test vector should deserialize");
+
+        // This should be ok
+        ready_verifier_service
+            .call(Arc::new(block.clone()))
+            .await
+            .map_err(|e| eyre!(e))?;
+
+        // Change nonce to something invalid
+        block.header.nonce = [0; 32];
+
+        /// Error: invalid equihash solution for BlockHeader
```

Comment:
We shouldn't try putting error messages in spans, instead of using `unwrap_err` you should use `expect_err` and put this comment as the message in `expect_err`, though you should probably change it to be something like `expect_err("expected the equihash solution to be invalid")`

Reply:
Thanks, fixed that.

Reply:
I think you might have pushed the wrong code, it looks like its still using `unwrap_err` right now

Reply:
do'h, sorry, fixed now :)